### PR TITLE
Remove list_types_tuple

### PR DESCRIPTION
**Pull request by mrocklin**
This removes `utils.list_types_tuple`, which was incorrect in that it contained the function `np.array`, which isn't a type.

Then we break apart the iloc code with an implementation that I hope also supports numpy arrays.

### CODE REVIEW

**Comment by mrocklin** [comment]
Previously calling `iloc` with a list would fail.

**Comment by mrocklin** [comment]
cc @zronaghi

File: python/cudf/utils/utils.py [hunk 1]
```diff
@@ -129,9 +129,6 @@ def standard_python_slice(len_idx, arg):
     return start, stop, step, slice_length
 
 
-list_types_tuple = (list, np.array)
```
Comment (mrocklin) [inline python/cudf/utils/utils.py]: FWIW, I think that we should avoid indirections like this in the future.  I think that they hide details where they're needed, and obfuscate the code without much advantage.

**Comment by mrocklin** [comment]
@zronaghi ran into it.  I'm not able to say how critical it is.

**Comment by zronaghi** [comment]
Not critical for me, working on gathering more data for the model.

**Comment by kkraus14** [comment]
Closing this as we're merging a general iloc update in #1184

**Pull request closed without merging**